parsers/FR.py

```diff
@@ -42,44 +46,63 @@ def is_not_nan_and_truthy(v) -> bool:
     return bool(v)
 
 
-@refetch_frequency(timedelta(days=1))
-def fetch_production(
-    zone_key: str = "FR",
+def get_dataset_from_datetime(target_datetime: datetime) -> str:
+    """Returns the dataset to query based on the target_datetime. The real-time API returns no values for target datetimes older than 9 months and we need to query the consolidated dataset."""
+    if target_datetime < arrow.now(tz="Europe/Paris").shift(months=-9):
+        # API called is Données éCO2mix régionales consolidées et définitives for datetimes that are more than 9 months in the past
+        dataset = DATASET_CONSOLIDATED
+    else:
+        dataset = DATASET_REAL_TIME
+    return dataset
+
+
+def get_data(
     session: Optional[Session] = None,
     target_datetime: Optional[datetime] = None,
-    logger: Logger = getLogger(__name__),
-) -> list:
+) -> pd.DataFrame:
+    """Returns a DataFrame with the data from the API."""
     if target_datetime:
-        to = arrow.get(target_datetime, "Europe/Paris")
+        target_datetime = arrow.get(target_datetime).replace(tzinfo="Europe/Paris")
     else:
-        to = arrow.now(tz="Europe/Paris")
+        target_datetime = arrow.now(tz="Europe/Paris")
```

Comment:
This is technically a type regression. As we define target_datetime to be Optional Datetime we can not assing a arrow object to it. We either need to remove the use of arrow, return the datetime instead of the arrow object or use a temporary variable like before.